### PR TITLE
Send DDNet version early in the connection process

### PR DESCRIPTION
**Pull request by heinrich5991**
This gets rid of the problem that we don't know whether we should send
full snapshots to clients because they haven't told us about them being
DDNet yet.

### CODE REVIEW

**Comment by heinrich5991** [comment]
After I'm through with changing everything, I'm not sure I like it anymore. Suggestions welcome.

**Comment by def-** [comment]
> I'm not sure I like it anymore.

Why not? On a quick glance it looks fine and should fix some problems with the version being only known later.

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -1218,6 +1229,79 @@ void CGameContext::OnClientEngineDrop(int ClientID, const char *pReason)
...
+	}
+	for(int i = 0; i < MAX_CLIENTS; i++)
+	{
+		if(m_apPlayers[i] && Score()->PlayerData(i)->m_CurrentTime > 0)
+		{
+			CNetMsg_Sv_PlayerTime Msg;
+			Msg.m_Time = Score()->PlayerData(i)->m_CurrentTime * 100;
+			Msg.m_ClientID = i;
+			Server()->SendPackMsg(&Msg, MSGFLAG_VITAL, ClientID);
+			//also send its time to others
```
Comment (Learath2) [inline src/game/server/gamecontext.cpp:1274]: Duplicate comment broken spacing. Also maybe move this into `SendTime` like all the others?

**Pull request merged**